Digitall.Warehouse.Application/Behaviors/LoggingBehavior.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Digitall.Warehouse.Domain.Shared;
+using MediatR;
+using Microsoft.Extensions.Logging;
+
+namespace Digitall.Warehouse.Application.Behaviors;
+
+public class LoggingBehavior<TRequest, TResponse> : IPipelineBehavior<TRequest, TResponse>
+{
+    private readonly ILogger<LoggingBehavior<TRequest, TResponse>> _logger;
+
+    public LoggingBehavior(ILogger<LoggingBehavior<TRequest, TResponse>> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task<TResponse> Handle(
+        TRequest request,
+        RequestHandlerDelegate<TResponse> next,
+        CancellationToken cancellationToken)
+    {
+        var commandName = request!.GetType().Name;
+
+        try
+        {
+            _logger.LogInformation(
+                "Handling command @{CommandName} with request {Command}", commandName, request);
+
+            var response = await next();
+
+            if (request.GetType().IsSubclassOf(typeof(IRequest<Result>)))
+            {
+                _logger.LogInformation("Command @{CommandName} handled.", commandName);
```

Comment:
I would suggest using high-performance logging if you are not familiar here is some video with benchmarks and example. (probably your VS is complaining and suggest to use the same)
https://www.youtube.com/watch?v=a26zu-pyEyg&t=2s